pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -591,15 +591,20 @@ func (s *DiscoveryServer) ProxyUpdate(clusterID, ip string) {
 	}
 
 	s.pushQueue.Enqueue(connection, &model.PushRequest{
-		Full:  true,
-		Push:  s.globalPushContext(),
-		Start: time.Now(),
+		Full:   true,
+		Push:   s.globalPushContext(),
+		Start:  time.Now(),
+		Reason: []model.TriggerReason{model.ProxyUpdate},
 	})
 }
 
 // AdsPushAll will send updates to all nodes, for a full config or incremental EDS.
 func AdsPushAll(s *DiscoveryServer) {
-	s.AdsPushAll(versionInfo(), &model.PushRequest{Full: true, Push: s.globalPushContext()})
+	s.AdsPushAll(versionInfo(), &model.PushRequest{
```

Comment:
I think currently this is only used by Debug which is manual push. Can we change and pass PushRequest so that debug can call this with right reason for example "Debug" , set rather than "Unknown" reason ?

Reply:
I made this debug